From fa4dbf094ae3d80adf65a912a6c4464531c41ac8 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Sun, 27 Mar 2016 15:03:29 +0200
Subject: [PATCH 0510/1345] dts: a3700: usb: gadget: update dts files for USB
 device support

commit  c548d1318ebbf9d59b479369447c433636b39ccd from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Added u3d (USB3) and udc (USB2) nodes in dts for USB device.

Enable USB device in board level:
1. enable both u3d and udc nodes
2. disable relevant USB3 host node

Example (replace USB3 Host with USB Device):

u3d@50000 {
	status = "okay";
};

udc@54100 {
	status = "okay";
};

/* CON31 */
usb3@58000 {
	status = "disabled";
	usb-phy = <&usb3_phy>;
};

Change-Id: Id878443a7fa9032354911fc2943e32ab115697be
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32812
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts    |    8 ++++++++
 arch/arm64/boot/dts/marvell/armada-3720-rz-db.dts |    8 ++++++++
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi      |   16 ++++++++++++++++
 3 files changed, 32 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index c012fa2..6f78653 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -93,6 +93,14 @@
 
 				enable-active-high;
 			};
+
+			u3d@50000 {
+				status = "disabled";
+			};
+
+			udc@54100 {
+				status = "disabled";
+			};
 		};
 
 		reg_sata: pwr-sata {
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-rz-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-rz-db.dts
index f78f87f..4a9af0a 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-rz-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-rz-db.dts
@@ -128,6 +128,14 @@
 				};
 			};
 
+			u3d@50000 {
+				status = "disabled";
+			};
+
+			udc@54100 {
+				status = "disabled";
+			};
+
 			/* CON31 */
 			usb3@58000 {
 				status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index 544bce7..ee24abf 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -317,6 +317,22 @@
 
 			};
 
+			u3d@50000 {
+				compatible = "marvell,mvebu-u3d";
+				/* 0: 0x50000: USB 3.0 Device port 0: DEV_INFO_REG(0:15 - version_id) */
+				reg = <0x50000 0x2000>;
+				interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&sgateclk 17>;
+				status = "disabled";
+			};
+
+			udc@54100 {
+				compatible = "marvell,mv-udc";
+				reg = <0x54100 0x2000>;
+				interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&sgateclk 17>;
+				status = "disabled";
+			};
 			usb3: usb@58000 {
 				compatible = "marvell,armada3700-xhci",
 				"generic-xhci";
-- 
1.7.9.5

